### PR TITLE
Make locales extraction script work with and without token

### PR DESCRIPTION
**Pull request by diox**
Helps while we're waiting to have fully automated l10n extraction (https://github.com/mozilla/addons-server/issues/9112)

### CODE REVIEW

**Comment by diox** [comment]
Please do test locally. It should work from the container, but do note that it will attempt to checkout to `master` first.

File: scripts/extract-l10n.sh [hunk 1]
```diff
@@ -8,10 +8,10 @@
 #   - Update your code
 #   - Extract new strings and push to the .po files
 #
-# If you're running the script manually please make sure to expose
+# For the fully automated experience make sure to expose
 # the following variables to the environment:
 #   - GITHUB_TOKEN (to the github token of addons-robot,
-#                   talk to tofumatt or cgrebs)
+#                   talk to @diox or @muffinresearch)
 #   - TRAVIS_REPO_SLUG="mozilla/addons-server"
```
Comment (eviljeff) [inline scripts/extract-l10n.sh:15]: Can't we provide this as a default in the file?

File: scripts/extract-l10n.sh [hunk 2]
```diff
@@ -8,10 +8,10 @@
...
 #   - TRAVIS_BRANCH="master"
```
Comment (eviljeff) [inline scripts/extract-l10n.sh:16]: This doesn't appear to be used at all (despite the erroring if it's not set).

File: scripts/extract-l10n.sh [hunk 3]
```diff
@@ -8,10 +8,10 @@
...
```
Comment (eviljeff) [inline scripts/extract-l10n.sh:17]: In `init_environment` it does `git checkout master` which depends on you having an up-to-date local master.  Can we just `git checkout upstream/master` instead?   (Even better if we can always checkout from the repo in `TRAVIS_REPO_SLUG`)

File: scripts/extract-l10n.sh [hunk 4]
```diff
@@ -98,21 +100,23 @@ function create_auto_pull_request {
...
 init_environment
 
 extract_locales
 
-commit_and_push
+git_commit
+
+if [ "GITHUB_TOKEN" = "" ]
+then
+    echo "No Github token provided, you'll have to push the branch and create the pull request yourself."
```
Comment (eviljeff) [inline scripts/extract-l10n.sh:111]: I understand why we can't create the pull request without the github API access but we could still push the branch like https://github.com/mozilla/addons-frontend/pull/9375/files#diff-cae1c8c05fe95719db14844a8f236ab1R76

**Comment by eviljeff** [comment]
Anything that can be done to have https://github.com/mozilla/addons-frontend/pull/9375 and this work in the same way would be a big improvement - right now the two scripts seem to make a lot of assumptions about git checkouts and branches and it's hard to remember the differences.

**Comment by diox** [comment]
> I tried but it fails on `jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'load'.` - which I thought was fixed in another commit?

It was! which file was this ? Are you sure your `master` was up to date ?

**Comment by eviljeff** [comment]
> 
> 
> > I tried but it fails on `jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'load'.` - which I thought was fixed in another commit?
> 
> It was! which file was this ? Are you sure your `master` was up to date ?

Forgot to do `fetch upstream` first.  (can we do that too?).  

It completed and then said:
`
Cleaning out obsolete messages.  See bug 623634 for details.
done.

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'olympia@0eca485b3639.(none)')
`
- this is being run in the container so does it have all the relevant git details?

**Comment by diox** [comment]
Ah. Ok, so I should probably not even do the `commit` in that mode then I think, to be safe.

File: scripts/extract-l10n.sh (hunk 2 above)
Comment (diox) [reply scripts/extract-l10n.sh:16]: This is purely if used in travis to make sure it's not running against branches other than master

File: scripts/extract-l10n.sh (hunk 1 above)
Comment (diox) [reply scripts/extract-l10n.sh:15]: As `TRAVIS_BRANCH` this is automatically present in travis

File: scripts/extract-l10n.sh (hunk 3 above)
Comment (diox) [reply scripts/extract-l10n.sh:17]: mmm in the case of travis, `upstream` doesn't exist as the remote... I'm going to try to steal some logic from https://github.com/mozilla/addons-frontend/blob/master/bin/run-l10n-extraction for this, pulling from the remote directly.

**Comment by diox** [comment]
Going to rework this to use the same logic we use on the frontend.

File: scripts/extract-l10n.sh (hunk 1 above)
Comment (eviljeff) [reply scripts/extract-l10n.sh:15]: I'm thinking for local use so we don't have to define extra variables to make it run.

File: scripts/extract-l10n.sh (hunk 2 above)
Comment (eviljeff) [reply scripts/extract-l10n.sh:16]: okay, that makes sense.

File: scripts/extract-l10n.sh (hunk 1 above)
Comment (diox) [reply scripts/extract-l10n.sh:15]: Yeah I'll make sure no variables are needed for manual usage

**Pull request closed without merging**